File: app/src/main/java/com/rara/moviecatalog/adapter/MovieAdapter.java
package com.rara.moviecatalog.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.rara.moviecatalog.R;
import com.rara.moviecatalog.model.Movie;

import java.util.ArrayList;

public class MovieAdapter extends BaseAdapter {
    private Context context;
    private ArrayList<Movie>movies;

    public MovieAdapter (Context context){
        this.context = context;
        movies = new ArrayList<>();
    }

    public void setMovies(ArrayList<Movie> movies) {
        this.movies = movies;
    }

    @Override
    public int getCount() {
        return movies.size();
    }

    @Override
    public Object getItem(int i) {
        return movies.get(i);
    }

    @Override
    public long getItemId(int i) {
        return i;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        if (view == null){
            view = LayoutInflater.from(context).inflate(R.layout.list_item, viewGroup, false);
        }
        ViewHolder viewHolder = new ViewHolder(view);
        Movie movie = (Movie) getItem(i);
        viewHolder.bind(movie);
        return view;
    }

    private class ViewHolder {
        private TextView tvJudul;
        private ImageView ivGambar;

        ViewHolder(View view) {
            tvJudul = view.findViewById(R.id.tvJudul);
            ivGambar = view.findViewById(R.id.ivGambar);
        }

        public void bind(Movie movie) {
            tvJudul.setText(movie.getJudul());
            ivGambar.setImageResource(movie.getPoster());
        }
    }
}

File: app/src/main/java/com/rara/moviecatalog/movie/MovieAdapter.kt
package com.rara.moviecatalog.movie

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.rara.moviecatalog.R
import com.rara.moviecatalog.api.ApiRepository
import com.rara.moviecatalog.model.Movie
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.list_item.view.*

class MovieAdapter(
    private val context: Context,
    private val listMovie: MutableList<Movie>,
    private val onClickListener: (Movie) -> Unit

) : RecyclerView.Adapter<MovieAdapter.ViewHolder>() {

    fun setData(movie: List<Movie>) {
        listMovie.clear()
        listMovie.addAll(movie)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = ViewHolder(
        LayoutInflater.from(context).inflate(R.layout.list_item, parent, false)
    )

    override fun getItemCount(): Int = listMovie.size

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(context, listMovie[position], onClickListener)
    }

    class ViewHolder(override val containerView: View) : RecyclerView.ViewHolder(containerView),
        LayoutContainer {
        fun bind(context: Context, movie: Movie, onClickListener: (Movie) -> Unit) {
            itemView.tvJudul.text = movie.title
            Glide.with(context).load(ApiRepository.BASE_IMAGE + movie.poster_path)
                .into(itemView.ivGambar)
            containerView.setOnClickListener { onClickListener(movie) }
        }

    }

}